src/Public/OpenApi.ps1

```diff
@@ -1165,113 +1997,513 @@ An array of other int/string/etc properties wrap up as an object.
+
+        if ($Required.IsPresent ) {
+            $param.required = $Required.ToBool()
+        }
+
+        if ($Deprecated.IsPresent ) {
+            $param.deprecated = $Deprecated.ToBool()
+        }
+
+        if ($Nullable.IsPresent ) {
+            $param.meta['nullable'] = $Nullable.ToBool()
+        }
+
+        if ($WriteOnly.IsPresent ) {
+            $param.meta['writeOnly'] = $WriteOnly.ToBool()
+        }
+
+        if ($ReadOnly.IsPresent ) {
+            $param.meta['readOnly'] = $ReadOnly.ToBool()
+        }
+
+        if ($Example ) {
+            $param.meta['example'] = $Example
+        }
+
+        if ($UniqueItems.IsPresent ) {
+            $param.uniqueItems = $UniqueItems.ToBool()
+        }
+
+        if ($Default) {
+            $param.default = $Default
+        }
+
+        if ($MaxItems) {
+            $param.maxItems = $MaxItems
+        }
+
+        if ($MinItems) {
+            $param.minItems = $MinItems
+        }
+
+        if ($MinProperties) {
+            $param.minProperties = $MinProperties
+        }
+
+        if ($MaxProperties) {
+            $param.maxProperties = $MaxProperties
+        }
+
+        if ($Xml) {
+            $param.xml = $Xml
+        }
+
+        $collectedInput = [System.Collections.Generic.List[hashtable]]::new()
+    }
+    process {
+        if ($ParamsList) {
+            $collectedInput.AddRange($ParamsList)
+        }
     }
 
-    return $param
+    end {
+        if ($collectedInput) {
+            $collectedInput + $param
+        } else {
+            return $param
+        }
+    }
 }
 
+
+<#
+.SYNOPSIS
+Creates a new OpenAPI object combining schemas and properties.
+
+.DESCRIPTION
+Creates a new OpenAPI object combining schemas and properties.
+
+.PARAMETER Type
+Define the type of validation between the objects
+oneOf – validates the value against exactly one of the subschemas
+allOf – validates the value against all the subschemas
+anyOf – validates the value against any (one or more) of the subschemas
+
+.PARAMETER Schemas
+An array of schemas or properties
+
+.PARAMETER Discriminator
+When request bodies or response payloads may be one of a number of different schemas, a discriminator object can be used to aid in serialization, deserialization, and validation.
+The discriminator is a specific object in a schema which is used to inform the consumer of the specification of an alternative schema based on the value associated with it.
+
+.EXAMPLE
+Add-PodeOAComponentSchema -Name 'Pets' -Schema (  New-PodeOAOf  -Type OneOf -Schema @( 'Cat','Dog') -Discriminator "petType")
+
+
+.EXAMPLE
+Add-PodeOAComponentSchema -Name 'Cat' -Schema (
+        New-PodeOAOf  -Type AllOf -Schema @( 'Pet', ( New-PodeOAObjectProperty -Properties @(
+                (New-PodeOAStringProperty -Name 'huntingSkill' -Description 'The measured skill for hunting' -Enum @(  'clueless', 'lazy', 'adventurous', 'aggressive'))
+                ))
+        ))
+#>
+function New-PodeOAOfProperty {
```

Comment:
I agree on Merge but I think is better to keep the Of in the name 
Merge-PodeOAOfProperty

Reply:
My only concern with keeping "Of" in the name is it sounds like it's merging multiple "Of" properties, rather than just merging multiple general properties.

If "Of" in OpenAPI is it's own kind of property, I'm happy to keep it using the "New" verb.

Reply:
In some way I agree with you I remove Of